lightning/src/events/bump_transaction.rs

```diff
@@ -700,8 +700,21 @@ where
 		&self, claim_id: ClaimId, package_target_feerate_sat_per_1000_weight: u32,
 		commitment_tx: &Transaction, commitment_tx_fee_sat: u64, anchor_descriptor: &AnchorDescriptor,
 	) -> Result<(), ()> {
+		// Our commitment transaction already has fees allocated to it, so we should take them into
+		// account. We compute its feerate and subtract it from the package target, using the result
+		// as the target feerate for our anchor transaction. Unfortunately, this results in users
+		// overpaying by a small margin since we don't yet know the anchor transaction size, and
+		// avoiding the small overpayment only makes our API even more complex.
+		let commitment_tx_sat_per_1000_weight: u32 = compute_feerate_sat_per_1000_weight(
+			commitment_tx_fee_sat, commitment_tx.weight() as u64,
+		);
+		let anchor_target_feerate_sat_per_1000_weight = core::cmp::max(
+			package_target_feerate_sat_per_1000_weight - commitment_tx_sat_per_1000_weight,
```

Comment:
IIUC here we’re subtracting the commitment transaction feerate sat per 1000 weight from the anchor feerate sat per 1000 weight. If the commitment transaction is 10 times bigger than the child, the anchor feerate might not be sufficient to meet efficient propagation and selection in network mempools.

I think this should rather go to 1) compute the commitment transaction absolute fee with its current feerate, 2) compute the commitment transaction absolute fee with `ConfirmationTarget::HighPriority`, 3) subtract result one from result two, 4) select enough UTXO value to meet the gap _assuming_ some anchor transaction number of inputs (as it’s the variable, all other transaction fields are invariant).

So I understand the concern laid about “we don’t yet know the anchor transaction size” though I would rather go the direction of modifying `select_confirmed_utxos` to take a `satoshis_value_to_complete` and parameterize with `per_additional_feerate_sat_per_1000_weight`.

Current state of code sounds to me we might propagate time-sensitive commitment transaction that broadcast feerate inefficient txn, like `FEERATE_FLOOR_SATS_PER_KW` is only a belt-and-suspender for min_relay_tx_fee iirc.

Reply:
>  thus even if the commitment transaction is huge,

In a world where an adversary can route `update_add_htlc` over your node to inflate the size of your commitment transaction, I think we would be better to be careful to have precise bounds on how much we overshoot (and a drain vector on our fee-bumping reserves).

I think this would benefit to have a functional test to get in sync on the feerate math here, and the overshoot bounds.

Reply:
> Currently we simply ask the user to match the sought feerate, which may result in substantially under-shooting the sought feerate if the commitment transaction is large.

This is correct from looking at code at (`8e2b70d9`) neither current `CoinSelectionSource::select_confirmed_utxos`’s `target_feerate_sat_per_1000_weight` or `BumpTransactionEventHandler::build_anchor_tx` document clearly what is included in their respective `target_feerate_sat_per_1000_weight` from what I can grep. Note how it can be confusing as you have `HTLCResolution::target_feerate_sat_per_1000_weight` and `ChannelClose::package_target_feerate_per_1000_weight`. 

So yes if you don’t include the commitment weight in your weight computation I agree you will undershoot and your transaction won’t propagate (fuck with all my respect for Wilmer hard work during the past year here, this API gently sucks in its current state for something which is safety critical).

> You make a good point that overshooting may exhaust our available funds for anchoring, but I'd much rather that than substantially undershoot our fee estimator and end up not getting our transactions confirmed.

Well I observe again than fee-bumping reserves are supposed to be under your local LN node control though note the commitment weight is not due to how forwarding works (or at least unless in the bounds of your channel policy parameters…). I don’t know what is worst, this is depend the deployment.

Thanks for the math, I’ll argue the API suggested above with code-enforced bounds on the commitment weight is better, so now I'm thinking we might have to leak the channel policy parameters there in `CoinSelectionSource` so I think conversation can be moved in its own issue / PR.